### PR TITLE
[12.0][IMP] openupgrade_records: detect deleted domains

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
The detection of deleted contexts cannot be pushed here, but locally can be done by a little trick.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
And what about context?

File: odoo/addons/openupgrade_records/lib/compare.py [hunk 1]
```diff
@@ -287,11 +287,22 @@ def compare_xml_sets(old_records, new_records):
 
     # direct match
     match_fields = ['module', 'model', 'name']
+    modified_records = []
     for column in copy.copy(old_records):
         found = search(column, new_records, match_fields)
         if found:
             old_records.remove(column)
             new_records.remove(found)
+            if column['domain'] != found['domain'] and \
```
Comment (pedrobaeza) [inline odoo/addons/openupgrade_records/lib/compare.py]: Can't all these ifs be simplified a bit? And why you need to do something similar several times? Or at least they can be extracted to a method.

**Comment by MiquelRForgeFlow** [comment]
> And what about context?

I said before:
> The detection of deleted contexts cannot be pushed here, but locally can be done by a little trick.

File: odoo/addons/openupgrade_records/lib/compare.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply odoo/addons/openupgrade_records/lib/compare.py]: ok

**Comment by pedrobaeza** [comment]
But I don't get what that means... and why it doesn't work the same as with domain?

**Comment by MiquelRForgeFlow** [comment]
Because in the domain case, the implicit default domain is `False` and the explicit domain is `[]`. Thus, we can distinguish between them.
In the case of context, both the implicit and the explicit default context is `{}`. The trick I do locally is to change the default implicit context by `{ }`.

**Comment by pedrobaeza** [comment]
But if I get it correctly, it doesn't matter: what you try to see is if there's a change in that context. It doesn't matter that implicitly an empty context is the same as `{}`. If on N-1 you have empty context and in N `{}` context, you don't need to do anything.

**Comment by MiquelRForgeFlow** [comment]
I don't want to check when there is a change in a context. That would give us a lot of irrelevant data. What I want to detect is only when in old version you have a context different from `{}` and in new version you don't have an explicit context. The problem is: there is no way to distinguish between a context explicit declared as `{}` and a not declared context.

**Comment by pedrobaeza** [comment]
OK, now I see the problem. Thanks for the info.

**Comment by pedrobaeza** [comment]
Is this finished?

**Review by StefanRijnhart — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
Yeah, it's finished. It's just that I was waiting for Martin merging my related PRs in odoo to avoid doing merge of the two commits marked as "temporal". Do you prefer to wait or proceed to merge?

**Comment by pedrobaeza** [comment]
Is this detecting missing context then?

As they don't hurt even if fixed upstream, I would say to go on.

**Comment by MiquelRForgeFlow** [comment]
No, this doesn't detect context, as I explained before. But locally I run a detection of the context case, and the purchase commit case was the unique context case I found.

**Comment by pedrobaeza** [comment]
OK, I merge this, but can't you put anything for doing it work that not being done locally?

**Pull request merged**